fmom/ops_test.go

```diff
@@ -44,6 +44,10 @@ func deepEqual(p1, p2 P4) bool {
 	return Equal(p1, p2)
 }
 
+func deepEqualP3(p1, p2 r3.Vec) bool {
+	return p1.X == p2.X && p1.Y == p2.Y && p1.Z == p2.Z
```

Comment:
```suggestion
	return p1 == p2
```
should just work.

and thus, `deepEqualP3` doesn't carry too much of its weight.
I'd just drop it.

Reply:
Ha ok, nice! 

One curiosity question on this: I though there is no operator overloading in go ... how `obj1 == obj2` can work for a "user-defined" structure?

Reply:
`==` and `!=` work for all values (except slices, the equality operator isn't defined for those: that's why one can't use slices as a key of a `map`).
so they do work nicely for `struct` values: it's just the field/field comparison (and recursively)

Reply:
(that's not the same for other operators such as `<`, `>` or `!`)